sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/DefaultConfigProperties.java

```diff
@@ -59,11 +66,27 @@ private DefaultConfigProperties(
     environmentVariables.forEach(
         (name, value) -> config.put(name.toLowerCase(Locale.ROOT).replace('_', '.'), value));
     systemProperties.forEach(
-        (key, value) -> config.put(normalize(key.toString()), value.toString()));
+        (key, value) -> {
+          Object valueToWrite = getValueToWriteAppendingResourceAttributes(config, key, value);
```

Comment:
Shouldn't we add this kind of transformation to all places where config maps get merged? Perhaps it'd make sense to have a 
```java
Map<String, String> merge(Map<String, String> map1, Map<String, String> map2, Function<String, String> keyNormalizer)
```
function that could be called in all places where we merge configs.

Reply:
Conceptually, it probably does make sense to do this or something similar for all maps. Like I said, I wanted to be strategic/surgical with this change. I haven't seen the spec addressing this for things outside of resource attributes, so I dunno.

Do we know what other config items contain maps? I'm not sure where to find that offhand...